### DISCUSSION TITLE
Add ability to save compressed binary scenes at export.

### DISCUSSION THREAD

**Discussion by Koalamana9** [godotengine/godot-proposals#8229] [Editor]
--

**Comment by Calinou** [comment]
Text resources are already converted to binary on export using the **Editor > Export > Convert Text Resources to Binary** project setting, which is enabled by default.

**Reply by ThunderFD** [reply]
thanks to this setting they are binary, yes, but they are not compressed as they would be when saving from gdscript like so:
```
ResourceSaver.save(resource, path, ResourceSaver.FLAG_COMPRESS)
```

for some resources this can actually enlarge .tres when they are converted to .res during export.

not sure if there is a flag to lightly compress these? or all binary resources? would be an easy way to get some export size improvements.

**Reply by Calinou** [reply]
See these PRs:

- https://github.com/godotengine/godot/pull/103287
- https://github.com/godotengine/godot/pull/103284
- https://github.com/godotengine/godot/pull/104571

These can substantially reduce the PCK size, at the cost of not being friendly to delta-encoded patch PCKs (since compression is not delta-friendly in general).
